.github/workflows/ci.yaml

```diff
@@ -64,36 +74,61 @@ jobs:
+        uses: docker/setup-buildx-action@v3
+
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: voxbox-${{ matrix.rubygem_puppet }}_${{ github.sha }}
+          path: /tmp
+
+      - name: Load Docker image
+        run: |
+          docker load --input /tmp/voxbox-${{ matrix.rubygem_puppet }}_${{ github.sha }}.tar
+          docker image ls -a
+
       - name: Clone voxpupuli/puppet-example repository
         uses: actions/checkout@v4
         with:
           repository: voxpupuli/puppet-example
 
       - name: Test container
         run: |
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile -T
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile lint
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile metadata_lint
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile strings:validate:reference
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile rubocop
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile syntax
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile spec
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile r10k:syntax
-          docker run --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile r10k:dependencies
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile -T
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile lint
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile metadata_lint
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile strings:validate:reference
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile rubocop
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile syntax
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile spec
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile r10k:syntax
+          docker run --user 1001 --rm  -v $(pwd):/repo ci/voxbox-${{ matrix.rubygem_puppet }}:${{ github.sha }} -f /Rakefile r10k:dependencies
```

Comment:
when we use the container locally, do we now have to set `--user`?

Reply:
no, you just can used it as always. at least on my mac it was working like this.
can you maybe test this on a non mac?

```
docker build -t voxbox-local .
cd path/to/puppet-example
docker run -it --rm -v $(pwd):/repo voxbox-local -f /Rakefile strings:validate:reference
```

Reply:
it could be, that one does need code like this in a compose file. but thats how this id done, afaik

```
    environment:
      - PUID=1001
      - PGID=1001
```

Reply:
build output: https://gist.github.com/bastelfreak/4cc2e9febafd4f40cf4a0769ad3c9672

and the rake task fails:

```
$ docker run -it --rm -v $(pwd):/repo voxbox-local -f /Rakefile strings:validate:reference
/usr/local/bundle/gems/io-event-1.6.5/lib/io/event/support.rb:27: warning: IO::Buffer is experimental and both the Ruby and C interface may change in the future!
WARN: Unresolved or ambiguous specs during Gem::Specification.reset:
      racc (>= 0)
      Available/installed versions of this gem:
      - 1.8.1
      - 1.6.2
      minitest (>= 5.1, ~> 5.4)
      Available/installed versions of this gem:
      - 5.24.1
      - 5.16.3
      drb (>= 0)
      Available/installed versions of this gem:
      - 2.2.1
      - 2.1.1
      mutex_m (>= 0)
      Available/installed versions of this gem:
      - 0.2.0
      - 0.1.2
WARN: Clearing out unresolved specs. Try 'gem cleanup <gem>'
Please report a bug if this causes problems.
rake aborted!
Errno::EACCES: Permission denied @ dir_s_mkdir - .yardoc (Errno::EACCES)
/usr/local/bundle/gems/yard-0.9.36/lib/yard/core_ext/file.rb:59:in `open!'
/usr/local/bundle/gems/yard-0.9.36/lib/yard/serializers/yardoc_serializer.rb:55:in `lock_for_writing'
/usr/local/bundle/gems/yard-0.9.36/lib/yard/registry_store.rb:202:in `lock_for_writing'
/usr/local/bundle/gems/yard-0.9.36/lib/yard/registry.rb:210:in `lock_for_writing'
/usr/local/bundle/gems/yard-0.9.36/lib/yard/cli/yardoc.rb:258:in `run'
/usr/local/bundle/gems/yard-0.9.36/lib/yard/cli/command.rb:14:in `run'
/usr/local/bundle/gems/puppet-strings-4.1.2/lib/puppet-strings.rb:48:in `generate'
/usr/local/bundle/gems/puppet-strings-4.1.2/lib/puppet-strings/tasks/validate.rb:29:in `block (4 levels) in <top (required)>'
/usr/local/bundle/gems/puppet-strings-4.1.2/lib/puppet-strings/tasks/validate.rb:21:in `block (3 levels) in <top (required)>'
/usr/local/bundle/gems/rake-13.2.1/exe/rake:27:in `<top (required)>'
Tasks: TOP => strings:validate:reference
(See full trace by running task with --trace)
```

Reply:
i cannot find how to prevent yard from creating this dir. also i read on the net, that it is common to run the container with `--user`? any opinions on this?